tests/draft4/ref.json

```diff
@@ -174,6 +174,35 @@
         ]
     },
     {
+        "description": "nested ref with other properties",
+        "schema": {
+            "definitions": {
+                "a": {
+                    "type": "number"
+                }
+            },
+            "properties": {
+                "foo": {
+                    "type": "number",
+                    "maximum": 5,
+                    "$ref": "#/definitions/a"
+                }
```

Comment:
Your understanding is correct as far as I undertsand. I haven't however reviewed this specific change.